src/dbus_api/pool/pool_3_8/api.rs

```diff
@@ -102,3 +102,50 @@ pub fn clevis_infos_property(f: &Factory<MTSync<TData>, TData>) -> Property<MTSy
         .emits_changed(EmitsChangedSignal::True)
         .on_get(get_pool_clevis_infos)
 }
+
+pub fn encrypt_pool_method(f: &Factory<MTSync<TData>, TData>) -> Method<MTSync<TData>, TData> {
+    f.method("EncryptPool", (), encrypt_pool)
+        // Optional key descriptions of key in the kernel keyring
+        // a: array of zero or more elements
+        // b: true if a token slot is specified
+        // i: token slot
+        // s: key description
+        //
+        // Rust representation: Vec<((bool, u32), String)>
+        .in_arg(("key_desc", "a((bu)s)"))
+        // Optional Clevis infos for binding on initialization.
+        // a: array of zero or more elements
+        // b: true if a token slot is specified
+        // i: token slot
+        // s: pin name
+        // s: JSON config for Clevis use
+        //
+        // Rust representation: Vec<((bool, u32), String, String)>
+        .in_arg(("clevis_info", "a((bu)ss)"))
+        // b: true if pool was newly encrypted
+        //
+        // Rust representation: bool
+        .out_arg(("results", "b"))
+        .out_arg(("return_code", "q"))
+        .out_arg(("return_string", "s"))
+}
+
+pub fn reencrypt_pool_method(f: &Factory<MTSync<TData>, TData>) -> Method<MTSync<TData>, TData> {
+    f.method("ReencryptPool", (), reencrypt_pool)
+        // b: true if successful
+        //
+        // Rust representation: bool
+        .out_arg(("results", "b"))
+        .out_arg(("return_code", "q"))
+        .out_arg(("return_string", "s"))
+}
+
+pub fn decrypt_pool_method(f: &Factory<MTSync<TData>, TData>) -> Method<MTSync<TData>, TData> {
```

Comment:
Can we consider calling this UnencryptPool rather than DecryptPool? decrypt generally at least connotes a slightly different thing.

Reply:
Do you mind if we solicit feedback on this? I do not know what will be clearer to the average user in this case. Even from a quick search of opinions across the internet, some people associate the verb "unencrypt" only with the state of being "unencrypted" (never having been encrypted) and not with any usage of the verb. Personally I don't think decrypt necessarily denotes something different as much as you can temporarily or persistently decrypt data. Temporarily decrypting seems to be the process of unlocking a device where persistently decrypting data seems to indicate changing the device to store unencrypted data.

Reply:
We decided in standup that this nomenclature is Ok.